task/k8s/task.go

```diff
@@ -184,44 +183,42 @@ func (t *Task) Read(ctx context.Context) error {
 
 func (t *Task) Delete(ctx context.Context) error {
 	logrus.Info("Deleting resources...")
+	steps := []common.Step{}
 	if t.Attributes.DirectoryOut != "" && t.Read(ctx) == nil {
-		os.Setenv("TPI_TRANSFER_MODE", "true")
-		os.Setenv("TPI_PULL_MODE", "true")
-		defer os.Unsetenv("TPI_TRANSFER_MODE")
-		defer os.Unsetenv("TPI_PULL_MODE")
-
-		logrus.Info("[1/7] Deleting Job...")
-		if err := t.Resources.Job.Delete(ctx); err != nil {
-			return err
-		}
-		logrus.Info("[2/7] Creating ephemeral Job to retrieve directory...")
-		if err := t.Resources.Job.Create(ctx); err != nil {
-			return err
-		}
-		logrus.Info("[3/7] Downloading Directory...")
-		if err := t.Pull(ctx, t.Attributes.Directory, t.Attributes.DirectoryOut); err != nil {
-			return err
+		env := map[string]string{
+			"TPI_TRANSFER_MODE": "true",
+			"TPI_PULL_MODE":     "true",
 		}
 
-		logrus.Info("[4/7] Deleting ephemeral Job to retrieve directory...")
-		if err := t.Resources.Job.Create(ctx); err != nil {
-			return err
-		}
-
-		os.Unsetenv("TPI_TRANSFER_MODE")
-		os.Unsetenv("TPI_PULL_MODE")
+		steps = []common.Step{{
+			Description: "Deleting Job...",
+			Action:      withEnv(env, t.Resources.Job.Delete),
+		}, {
+			Description: "Creating ephemeral Job to retrieve directory...",
+			Action:      withEnv(env, t.Resources.Job.Create),
+		}, {
+			Description: "Downloading Directory...",
+			Action: withEnv(env, func(ctx context.Context) error {
+				return t.Pull(ctx, t.Attributes.Directory, t.Attributes.DirectoryOut)
+			}),
+		}, {
+			// WTH?
+			Description: "Deleting ephemeral Job to retrieve directory...",
```

Comment:
Smoke tests fail if we replace `t.Resources.Job.Create` with `Delete`

Reply:
Resolving in favor of https://github.com/iterative/terraform-provider-iterative/issues/647.